app/src/main/java/com/example/bigowlapp/model/Schedule.java

```diff
@@ -111,4 +114,57 @@ public void setLocation(GeoPoint location) {
     public void setUserScheduleResponseMap(Map<String, UserScheduleResponse> userScheduleResponseMap) {
         this.userScheduleResponseMap = userScheduleResponseMap;
     }
+
+    @Exclude
+    public Map<String, Object> scheduleMemberResponseAttendanceMap(String memberId) {
+        UserScheduleResponse userScheduleResponse = this.userScheduleResponseMap.get(memberId);
+        Map<String, Object> map = new HashMap<>();
+        if (userScheduleResponse.getAttendance().didAttend()) {
+            map.put("responseText", "ATTENDED");
+            map.put("responseColor", Color.GREEN);
+            map.put("attendanceTime", userScheduleResponse.getAttendance().getAuthenticationTime().toDate().toString());
+        } else if (userScheduleResponse.getResponse() == Response.REJECT) {
+            map.put("responseText", "REJECTED");
+            map.put("responseColor", Color.RED);
+        } else if (userScheduleResponse.getResponse() == Response.NEUTRAL) {
+            map.put("responseText", "NO RESPONSE");
+            map.put("responseColor", Color.GRAY);
+        } else {
+            switch (scheduleCurrentState()) {
+                case SCHEDULED:
+                    map.put("responseText", "ACCEPTED");
+                    map.put("responseColor", Color.GREEN);
+                    break;
+                case ON_GOING:
+                    map.put("responseText", "PENDING");
```

Comment:
More than 30 minutes after the schedule start time, the schedule result should no more be "PENDING", because that is the time limit after which the system will stop searching for the user location. So you should make the schedule result as "MISSED" if after this time limit the user has not yet attended the schedule

Reply:
![image](https://user-images.githubusercontent.com/35667311/113355224-6ea40400-930e-11eb-8d43-7a0584bdf726.png)
this will fulfill what you requested. Thanks!